runtime/sema/type.go

```diff
@@ -291,6 +293,49 @@ func MustInstantiate(t ParameterizedType, typeArguments ...Type) Type {
 	)
 }
 
+func CheckParameterizedTypeInstantiated(
+	t ParameterizedType,
+	pos ast.HasPosition,
+	memoryGauge common.MemoryGauge,
+	report func(err error),
+) {
+	typeArgs := t.TypeArguments()
+	typeParameters := t.TypeParameters()
+
+	typeArgumentCount := len(typeArgs)
+	typeParameterCount := len(typeParameters)
+
+	if typeArgumentCount != typeParameterCount {
+		report(
+			&InvalidTypeArgumentCountError{
+				TypeParameterCount: typeParameterCount,
+				TypeArgumentCount:  typeArgumentCount,
+				Range: ast.NewRange(
+					memoryGauge,
+					pos.StartPosition(),
+					pos.EndPosition(memoryGauge),
+				),
+			},
+		)
+	}
```

Comment:
codecov reports this section is not covered by tests. Can you add a test(s) that could cover this part as well. e.g: initialized, but with invalid arg count?

Reply:
This was redundant as the check also happens in the `convertInstantiationType` function. So I removed it from here.